netbox_lifecycle/template_content.py

```diff
@@ -3,28 +3,33 @@
 from django.template import Template
 from netbox.plugins import PluginTemplateExtension
 
-from .models import hardware, contract
+from .models import hardware, contract, license
 
 
 class DeviceHardwareInfoExtension(PluginTemplateExtension):
     def right_page(self):
         object = self.context.get('object')
-        support_contract = contract.SupportContractAssignment.objects.filter(device_id=self.context['object'].id).first()
+        max_items_display = 5
         match self.kind:
             case "device":
+                licenses = license.LicenseAssignment.objects.filter(device_id=self.context['object'].id)[:max_items_display]
+                support_contract = contract.SupportContractAssignment.objects.filter(device_id=self.context['object'].id).first()
                 content_type = ContentType.objects.get(app_label="dcim", model="devicetype")
                 lifecycle_info = hardware.HardwareLifecycle.objects.filter(assigned_object_id=self.context['object'].device_type_id,
                                                                            assigned_object_type_id=content_type.id).first()
             case "module":
+                licenses = None
                 content_type = ContentType.objects.get(app_label="dcim", model="moduletype")
                 lifecycle_info = hardware.HardwareLifecycle.objects.filter(assigned_object_id=self.context['object'].module_type_id,
                                                                            assigned_object_type_id=content_type.id).first()
             case "devicetype" | "moduletype":
+                licenses = None
                 content_type = ContentType.objects.get(app_label="dcim", model=self.kind)
                 lifecycle_info = hardware.HardwareLifecycle.objects.filter(assigned_object_id=self.context['object'].id,
                                                                            assigned_object_type_id=content_type.id).first()
-        context = {'support_contract': support_contract, 'lifecycle_info': lifecycle_info}
-        return self.render('netbox_lifecycle/inc/support_contract_info.html', extra_context=context)
+        context = {'max_items_display': max_items_display, 'licenses': licenses, 'support_contract': support_contract,
+                   'lifecycle_info': lifecycle_info}
+        return self.render('netbox_lifecycle/inc/licenses_info.html', extra_context=context)
```

Comment:
At a glance, this looks to remove the lifecycle information.  Instead of removing this and including at the bottom, it is likely better if you only include the card as a new card and only on the device page (not on other pages).